pybase/region/client.py

```diff
@@ -171,32 +156,30 @@ def _send_request(self, rq, lock_timeout=10):
     #   4. A varint representing the length of the serialized ResponseMessage.
     #   5. The ResponseMessage.
     #
-    def _receive_rpc(self, call_id, rq, data=None, lock_timeout=10):
+    @staticmethod
+    def send_and_receive_rpc(client, call_id, rq, to_send):
+        thread_name = current_thread().name
+        sp = thread_name.split("_") # i.e. splitting "ThreadPoolExecutor-1_0"
+        pool_id = int(sp[1]) # thread number is now responsible for only using its matching socket
+
+        client.sock_pool[pool_id].send(to_send)
```

Comment:
Should we move the `send` into the `try`  below to be able use the same exception handling?

Reply:
yes